File: BACK/Traffic Police/dto/OwnershipTransferDto.go
package dto

import (
	"eUprava/trafficPolice/model"
	"encoding/json"
	"io"
	"time"
)

type OwnershipTransferDto struct {
	Id             string           `json:"id"`
	Vehicle        model.VehicleDTO `json:"vehicle"`
	OldOwner       model.OwnerDTO   `json:"oldOwner"`
	NewOwner       model.OwnerDTO   `json:"newOwner"`
	DateOfTransfer time.Time        `json:"dateOfTransfer"`
}

func (o *OwnershipTransferDto) ToJSON(w io.Writer) error {
	e := json.NewEncoder(w)
	return e.Encode(o)
}
func (o *OwnershipTransferDto) FromJSON(r io.Reader) error {
	e := json.NewDecoder(r)
	return e.Decode(o)
}

File: BACK/traffic-police/src/main/java/com/example/traffic_police/model/OwnershipTransferDTO.java
package com.example.traffic_police.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class OwnershipTransferDTO {
    private String id;
    private VehicleDTO vehicle;
    private OwnerDTO ownerOld;
    private OwnerDTO ownerNew;
    private LocalDateTime dateOfTransfer;
}